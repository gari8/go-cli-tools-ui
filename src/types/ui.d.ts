import { IconType } from "react-icons";

export interface IPageObject {
  name: string;
  url: string;
  icon?: IconType;
}
