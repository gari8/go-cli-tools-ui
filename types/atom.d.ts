import { NodeKey, AtomEffect, RecoilValue } from "recoil";

export interface AtomOptions<T> {
  key: NodeKey;
  default: RecoilValue<T> | Promise<T> | T;
  effects_UNSTABLE?: ReadonlyArray<AtomEffect<T>>;
  dangerouslyAllowMutability?: boolean;
}
