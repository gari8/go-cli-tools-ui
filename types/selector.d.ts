import { GetRecoilValue, RecoilValue } from "recoil";

export interface ReadOnlySelectorOptions<T> {
  key: string;
  get: (opts: { get: GetRecoilValue }) => Promise<T> | RecoilValue<T> | T;
  dangerouslyAllowMutability?: boolean;
}
