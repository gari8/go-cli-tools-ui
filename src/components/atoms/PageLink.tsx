import { IPageObject } from "~/types/ui";
import { Box, Flex, Text } from "@chakra-ui/react";
import { Icon } from "@chakra-ui/icons";
import { useRouter } from "next/router";

interface Props {
  page: IPageObject;
  selectedColor?: string;
  pointerColor?: string;
}

export default function PageLink({ page, selectedColor, pointerColor }: Props) {
  const router = useRouter();
  const isCurrentPage = (): boolean => router.pathname === page.url;
  const pointer = pointerColor ? pointerColor : "brand.main";
  const selected = selectedColor ? selectedColor : "brand.main";
  return (
    <Box p={1}>
      <Flex
        borderRadius={5}
        py={2}
        px={4}
        transition={"all .3s"}
        cursor={"pointer"}
        bg={isCurrentPage() ? selected : ""}
        _hover={{ bg: pointer, transition: "all .2s" }}
        onClick={() => router.push(page.url)}
      >
        <Flex flexDirection={"column"} justify={"center"} mx={2}>
          <Icon as={page.icon} />
        </Flex>
        <Text>{page.name}</Text>
      </Flex>
    </Box>
  );
}
