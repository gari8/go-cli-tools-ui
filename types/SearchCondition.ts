export default interface SearchCondition {
  searchWords: string[];
  author?: string;
  updatedAt?: Date;
  startedAt?: Date;
  stars?: number;
  tags?: string[];
}
