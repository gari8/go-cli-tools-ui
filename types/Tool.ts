export default interface Tool {
  id: string;
  title: string;
  author: string;
  description: string;
  githubUrl: string;
  binaries: string[];
}
