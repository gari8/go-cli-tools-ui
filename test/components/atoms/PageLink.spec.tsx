import { render } from "@testing-library/react";
import PageLink from "../../../src/components/atoms/PageLink";
import { IPageObject } from "../../../src/types/ui";

describe("PageLink", (): void => {
  const testPage: IPageObject = {
    name: "test",
    url: "/",
  };
  const useRouter = jest.spyOn(require("next/router"), "useRouter");
  test("is components", () => {
    useRouter.mockImplementationOnce(() => ({
      query: { pathname: "/" },
    }));
    const { asFragment } = render(<PageLink page={testPage} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
