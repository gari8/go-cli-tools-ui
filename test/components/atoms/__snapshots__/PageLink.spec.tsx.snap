// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLink is components 1`] = `
<DocumentFragment>
  <div
    class="css-5tigzz"
  >
    <div
      class="css-cezcce"
    >
      <div
        class="css-lyvtfg"
      >
        <svg
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <g
            stroke="currentColor"
            stroke-width="1.5"
          >
            <path
              d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
              fill="none"
              stroke-linecap="round"
            />
            <path
              d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
              fill="currentColor"
              stroke-linecap="round"
            />
            <circle
              cx="12"
              cy="12"
              fill="none"
              r="11.25"
              stroke-miterlimit="10"
            />
          </g>
        </svg>
      </div>
      <p
        class="chakra-text css-0"
      >
        test
      </p>
    </div>
  </div>
</DocumentFragment>
`;
